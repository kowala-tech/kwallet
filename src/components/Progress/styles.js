const styles = (theme) => ({
	root: {
		padding: theme.spacing.unit * 2,
		backgroundColor: "inherit",
		display: "flex",
		alignItems: "center",
		justifyContent: "center"
	}
});

export default styles;
