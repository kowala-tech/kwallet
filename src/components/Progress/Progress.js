import React from "react";
import PropTypes from "prop-types";
// Material UI Imports
import { withStyles } from "@material-ui/core/styles";
import CircularProgress from "@material-ui/core/CircularProgress";
// Component Related Imports
import styles from "./styles";

class Progress extends React.Component {

	render() {
		const { classes, ...rest } = this.props;

		return (
			<div className={classes.root}>
				<CircularProgress
					{...rest}
					color="secondary"/>
			</div>
		);
	}
}

Progress.propTypes = {
	classes: PropTypes.object.isRequired
};

export default withStyles(styles)(Progress);
