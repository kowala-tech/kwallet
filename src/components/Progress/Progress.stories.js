import React from "react";
import { storiesOf } from "@storybook/react";
// Story related component imports
import Progress from "./Progress";

storiesOf("Progress", module)
	.add("default", () =>
		<Progress />
	);
