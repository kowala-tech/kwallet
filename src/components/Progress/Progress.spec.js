import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
// Story related component imports
import Progress from "./Progress";

describe("Progress", () => {

	it("it should match the snapshot", () => {
		const component = toJson(
			shallow(
				<Progress/>
			)
		);
		expect(component).toMatchSnapshot();
	});

});
