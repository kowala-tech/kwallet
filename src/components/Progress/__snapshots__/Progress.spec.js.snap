// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress it should match the snapshot 1`] = `
<Progress
  classes={
    Object {
      "root": "Progress-root-1",
    }
  }
/>
`;
